apps/site/assets/ts/stop/components/DeparturesAndMap.tsx

```diff
@@ -0,0 +1,107 @@
+import React, { ReactElement, useState } from "react";
+import { Polyline } from "leaflet";
+import { chain } from "lodash";
+import { DirectionId, Route, Stop } from "../../__v3api";
+import { ScheduleWithTimestamp } from "../../models/schedules";
+import StopPageDepartures from "./StopPageDepartures";
+import StopMapRedesign from "./StopMapRedesign";
+import { useRoutesByStop } from "../../hooks/useRoute";
+
+interface DeparturesAndMapProps {
+  routes: Route[];
+  stop: Stop;
+  schedules: ScheduleWithTimestamp[];
+  lines: Polyline[];
+}
+
+const DeparturesAndMap = ({
+  /* eslint-disable @typescript-eslint/no-unused-vars */
+  routes,
+  stop,
+  schedules,
+  lines
+}: /* eslint-enable @typescript-eslint/no-unused-vars */
+DeparturesAndMapProps): ReactElement<HTMLElement> => {
+  const [state, setState] = useState<{
+    email: null | string;
+    password: null | string;
+  }>({ email: null, password: null });
+
+  const [departureInfo, setDepartureInfo] = useState<{
+    departureRoute: Route | null;
+    departureDirectionId: DirectionId | null;
+    departureSchedules: ScheduleWithTimestamp[] | null;
+  }>({
+    departureRoute: null,
+    departureDirectionId: null,
+    departureSchedules: null
+  });
+
+  const routesWithPolylines = useRoutesByStop(stop.id);
+
+  const polylines = chain(routesWithPolylines)
```

Comment:
This is already passed in to the component as `lines`.  Why is it being recalculated?

Reply:
Okay, the issue is you are using the wrong `PolyLine` import.
This is the reason why I like uniquely named structs.
The polyline type the map is expecting is from `leaflet/components/__mapdata`
You are just importing the `leaflet` one.  Changing imports should fix the issue and allow the removal of this recalculation.

Though you are probably best in calculating it here.  Since this is becoming it's own component, and nothing in the `StopPageRedesign` requires it.  Just remove the polyline calculation in the `StopPageRegesign` as well.